app/src/main/java/com/example/gamingex/snake/HelpScreenAbstract.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.gamingex.snake
+
+import com.example.gamingex.framework.Game
+import com.example.gamingex.framework.Input
+import com.example.gamingex.framework.Screen
+
+abstract class HelpScreenAbstract(game: Game): Screen(game) {
+
+    abstract val order: Int
+
+    fun update(deltaTime: Float, number: Int) {
+        val touchEvents = game.getInput().touchEvents
+        game.getInput().keyEvents
+        for (event in touchEvents) {
+            if (event.type == Input.Companion.TouchEvent.TOUCH_UP) {
+                if (event.x > 256 && event.y > game.getGraphics().getHeight()-64) {
+                    game.setScreen(screenFactory(number, game))
+                    if (Settings.soundEnabled) Assets.click.play(1f)
+                }
+            }
+        }
+    }
+
+    override fun update(deltaTime: Float) {
+        update(deltaTime, order)
+    }
+
+    fun presentAbstract(number: Int) {
+        val g = game.getGraphics()
+        val asset = when (number) {
+            1-> Assets.help1
+            2-> Assets.help2
+            3-> Assets.help3
```

Comment:
Looks kind of reversed, these should be defined and used in a child class not here because if you want to add another child class with `order == 4` you would then have to go back here and add it.

The proper abstraction would be to have method `abstract fun getHelpAsset()` which would be defined in the child classes if that is needed.

Reply:
Thanks, I`ll fix that right after I finish Tetris.